internal/controller/thanosquery_controller_test.go

```diff
@@ -110,6 +119,168 @@ var _ = Describe("ThanosQuery Controller", func() {
 
 				}, time.Minute*1, time.Second*10).Should(Succeed())
 			})
+
+			By("setting endpoints on the thanos query", func() {
+				svc := &corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "thanos-receive",
+						Namespace: ns,
+						Labels: map[string]string{
+							manifests.DefaultStoreAPILabel: manifests.DefaultStoreAPIValue,
+						},
+					},
+					Spec: corev1.ServiceSpec{
+						Ports: []corev1.ServicePort{
+							{
+								Name:       "grpc",
+								Port:       10901,
+								TargetPort: intstr.FromInt(10901),
+							},
+						},
+					},
+				}
+				Expect(k8sClient.Create(context.Background(), svc)).Should(Succeed())
+
+				controllerReconciler := &ThanosQueryReconciler{
+					Client: k8sClient,
+					Scheme: k8sClient.Scheme(),
+				}
+
+				_, err := controllerReconciler.Reconcile(ctx, reconcile.Request{
+					NamespacedName: typeNamespacedName,
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				EventuallyWithOffset(1, func() error {
+					sa := &corev1.ServiceAccount{}
```

Comment:
lets move the func in https://github.com/thanos-community/thanos-operator/blob/main/internal/controller/thanosreceive_controller_test.go#L332 to test/utils.go and reuse it to avoid duplication?

Reply:
Notation is diff